lib/providers/transit.js

```diff
@@ -0,0 +1,83 @@
+'use strict';
+
+const Vaulted = require('vaulted');
+const preconditions = require('conditional');
+const checkNotEmpty = preconditions.checkNotEmpty;
+
+// Default Vault connection options
+const VAULT_CONFIG = {
+  vault_host: Config.get('vault:host'),
+  vault_port: Config.get('vault:port'),
+  vault_ssl: Config.get('vault:tls')
+};
+
+/**
+ * The TransitProvider is responsible for retrieving secrets from Vault's transit secret backend.
+ * https://www.vaultproject.io/docs/secrets/transit/index.html
+ */
+class TransitProvider {
+  /**
+   * Create a new instance of a TransitProvider.
+   *
+   * @param {String} key - Key in Vault to use for decryption
+   * @param {String} token - Token to pass to Vault's API
+   * @param {Object} parameters - Arguments to pass to Vault's /transit/decrypt/ API
+   * @param {Object} parameters.ciphertext - Ciphertext to decrypt
+   */
+  constructor(key, token, parameters) {
+    checkNotEmpty(key, 'key is required');
+    checkNotEmpty(token, 'token is required');
+    checkNotEmpty(parameters, 'parameters is required');
+    checkNotEmpty(parameters.ciphertext, 'parameters.ciphertext is required');
+
+    this._key = key;
+    this._token = token;
+    this._parameters = parameters;
+
+    this._client = new Vaulted(VAULT_CONFIG);
+    this._client.api.mountEndpoints(this._client.config, 'transit', 'transit');
```

Comment:
This is resolved here: https://github.com/chiefy/vaulted/pull/96

Once it's merged to master you should update the dependency in `package.json`.